db/migrate/20180412124505_add_priority_score_to_discovery_rules.rb

```diff
@@ -0,0 +1,23 @@
+ class AddPriorityScoreToDiscoveryRules < ActiveRecord::Migration[4.2]
+  class DiscoveryRules < ActiveRecord::Base
+  end
+
+  def up
+    score = DiscoveryRules.distinct.order("priority asc").pluck(:priority).first
+    # To keep the same priority behavior before this change,
+    # set the priority score accordingly.
+    DiscoveryRules.reset_column_information
+    rules = DiscoveryRules.order("priority asc, name asc")
+    rules.each do |rule|
+      rule.update_attributes!(:priority => score)
+      score += 1
+      say "#{rule.name} priority is #{rule.priority}"
+    end
+    change_column :discovery_rules, :priority, :integer, unique: true, default: 0
```

Comment:
The `unique` option does not seem to be right. We actually want this column to be always present with no default: `null: false` - note the default is no longer here because we want it to fail when there is no priority. Then add index to guarantee consistency and faster search queries:

    add_index(:discovery_rules, :priority, unique: true)